contracts/Ovm_SpokePool.sol

```diff
@@ -141,17 +146,24 @@ contract Ovm_SpokePool is CrossDomainEnabled, SpokePool {
             WETH9(relayerRefundLeaf.l2TokenAddress).withdraw(relayerRefundLeaf.amountToReturn); // Unwrap into ETH.
             relayerRefundLeaf.l2TokenAddress = l2Eth; // Set the l2TokenAddress to ETH.
         }
-        IL2ERC20Bridge(
-            tokenBridges[relayerRefundLeaf.l2TokenAddress] == address(0)
-                ? Lib_PredeployAddresses.L2_STANDARD_BRIDGE
-                : tokenBridges[relayerRefundLeaf.l2TokenAddress]
-        ).withdrawTo(
-                relayerRefundLeaf.l2TokenAddress, // _l2Token. Address of the L2 token to bridge over.
+        // Handle custom SNX bridge which doesn't conform to the standard bridge interface.
+        if (relayerRefundLeaf.l2TokenAddress == 0x8700dAec35aF8Ff88c16BdF0418774CB3D7599B4)
```

Comment:
+1 LGTM.

As a part of this spoke pool upgrade (different PR), we could consider trying to make our spoke pool bridging logic more adaptable. Maybe we could use a delegatecalled adapter setup like mainnet, so we can replace the bridging logic easily without having to move the entire contract over.

Thoughts?

Reply:
yes definitely! Check out this [draft PR](https://github.com/across-protocol/contracts-v2/pull/208/files#diff-e3ad242c3b0904fa828d6a03174064c9133ca004a4ba53704b544b4ebda57826) where we can insert this custom SNX logic. I'm curious what you think of this draft architecture

Reply:
(if it doesn't link to the correct file, go down to the "OVM_SpokeAdapter")